include/picongpu/particles/particleToGrid/derivedAttributes/EnergyDensityCutoff.hpp

```diff
@@ -0,0 +1,65 @@
+/* Copyright 2013-2017 Axel Huebl, Rene Widera, Heiko Burau, Marco Garten
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/particles/particleToGrid/derivedAttributes/EnergyDensityCutoff.def"
+
+#include "picongpu/simulation_defines.hpp"
+#include "picongpu/algorithms/KinEnergy.hpp"
+
+namespace picongpu
+{
+namespace particleToGrid
+{
+namespace derivedAttributes
+{
+
+    HDINLINE float1_64
+    EnergyDensityCutoff::getUnit() const
+    {
+        constexpr float_64 UNIT_VOLUME = (UNIT_LENGTH * UNIT_LENGTH * UNIT_LENGTH);
+        return UNIT_ENERGY / UNIT_VOLUME;
+    }
+
+    template< class T_Particle >
+    DINLINE float_X
+    EnergyDensityCutoff::operator()( T_Particle& particle ) const
+    {
+        /* read existing attributes */
+        const float_X weighting = particle[weighting_];
+        const float3_X mom = particle[momentum_];
+        const float_X mass = attribute::getMass( weighting, particle );
+
+        constexpr float_X INV_CELL_VOLUME = float_X(1.0) / CELL_VOLUME;
+
+        /* value for energy cut-off */
+        constexpr float_X cutoff_energy = particles::ionization::thomasFermi::TF_CUTOFF_ENERGY;
```

Comment:
The functor `EnergyDensityCutoff` is part of the gerneric namespace `particleToGrid` but depends on values of an special ionization model *ThomasFermi*. 
This dependency is not mentioned in the documentation in the `*.def` file.
We need to pass the value of `particles::ionization::thomasFermi::TF_CUTOFF_ENERGY` somehow in the template interface or via constructor if the functor concept used following http://picongpu.readthedocs.io/en/latest/usage/particles.html?highlight=filter#manipulation-functors.